### DISCUSSION TITLE
Non riesco a creare un prodotto senza IVA standard, cioè lasciando "Ereditata dal cliente"

### DISCUSSION THREAD

**Discussion by d-one-lorenzo** [fattureincloud/api#402] [Q&A] [answered]
Buonasera. Sto provando la API `POST` per la creazione del prodotto, ma se non riempio l'oggetto `default_vat`, ottengo un errore `422 "Missing default_vat."`. Ho provato anche a passarlo NULL, cioè esattamente come mi viene ritornato dalla API `GET` per i prodotti che creo dalla vostra interfaccia lasciando l'IVA a "Ereditata dal cliente", ma ottengo lo stesso identico errore.

**Comment by emajo** [comment]
Ciao @d-one-lorenzo, molto strano questo comportamento, riesce a fornirci il json di creazione del prodotto così da fare qualche prova?

Grazie,
Emanuele.

**Reply by d-one-lorenzo** [reply]
Certo. La URL ovviamente è `POST https://api-v2.fattureincloud.it/c/---/products`, come body ho provato sia senza valore che con `null`:

```json
{
  "data": {
    "name": "Prodotto creato da API",
    "code": "TESTAPI00001",
    "net_price": 10,
    "net_cost": 20,
    "use_gross_price": 1,
    "gross_price": 30,
    "measure": "m",
    "category": "Prove",
    "description": "---",
    "notes": "---"
  }
}
```

```json
{
  "data": {
    "name": "Prodotto creato da API",
    "code": "TESTAPI00001",
    "net_price": 10,
    "net_cost": 20,
    "use_gross_price": 1,
    "gross_price": 30,
    "measure": "m",
    "category": "Prove",
    "description": "---",
    "notes": "---",
    "default_vat": null
  }
}
```

In entrambi i casi la risposta è un `422` con:

```json
{
  "error": {
    "message": "Missing default_vat.",
    "validation_result": null
  }
}
```

**Reply by d-one-lorenzo** [reply, marked as answer]
Ah, ora penso di aver capito. Con `"use_gross_price": 1` è obbligatorio inserire un'aliquota IVA esplicita, non può essere ereditata (e infatti dice così anche la vostra normale UI). Mi ero lasciato ingannare dal messaggio. Grazie lo stesso.
